packages/realm/src/app-services/Sync.ts

```diff
@@ -76,8 +76,9 @@ export namespace Sync {
     });
     app.internal.syncManager.setLoggerFactory(factory);
   }
-  export function getAllSyncSessions(user: User): SyncSession[] {
-    throw new Error("Not yet implemented");
+  export function getAllSyncSessions(user: User): SyncSession[] | null {
+    const allSessions = user.internal.allSessions();
+    return allSessions?.map((session) => new SyncSession(session)) ?? null;
```

Comment:
Even with the `Nullable<>` in place, you didn't need the `?.`. That meant that the SyncSession pointers *inside* the vector/Array may be null, not that the vector itself may be null.

Did the generated TS types cause an error/warning without the `?.`? If so, I think that would be a bug in the generator.

Reply:
Crap, looks like there was a bug! That should have been `(null | SyncSession)[]` but was rendered as `null | SyncSession[]` which is parsed as `null | (SyncSession[])`. Will fix.

Reply:
Fixed by https://github.com/realm/realm-js/commit/707f80a8b046eb35596672b65f73d45477c510a2